proto/elys/amm/query.proto

```diff
@@ -206,3 +206,31 @@ message QueryAMMPriceRequest {
   cosmos.base.v1beta1.Coin token_in = 1 [(gogoproto.nullable) = false];
   string                   discount   = 2 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
 }
+
+enum FilterType {
+  FilterAll = 0;
+	FilterPerpetual = 1;
+	FilterFixedWeight = 2;
+	FilterDynamicWeight = 3;
+}
+
+message QueryEarnPoolRequest {
+  repeated uint64 pool_ids = 1;
+  FilterType filter_type = 2;
+  cosmos.base.query.v1beta1.PageRequest pagination = 3;
+}
+
+message EarnPool {
+  repeated PoolAsset assets = 1 [(gogoproto.nullable) = false];
+  string pool_ratio   = 2 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string dex_apr   = 3 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string eden_apr   = 4 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string leverage_lp   = 5 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string perpetual   = 6 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string tvl   = 7 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string rewards   = 8 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+}
```

Comment:
@kenta-elys could you move those types to the types.proto file instead?

Reply:
yep